nmrfx-core/src/main/java/org/nmrfx/chemistry/relax/RelaxationData.java

```diff
@@ -173,48 +178,67 @@ public Map<String, String> getExtras() {
         return extras;
     }
 
-    public static  Map<Long, R1R2NOE>  assembleAtomData(Atom atom) {
+    public static Map<Long, EnumMap<relaxTypes, RelaxationData>> assembleAtomData(Atom atom) {
         var relaxData = atom.getRelaxationData();
-        Map<Long, R1R2NOE> dataMap = new HashMap<>();
+        Map<Long, EnumMap<relaxTypes, RelaxationData>> dataMap = new HashMap<>();
         Set<Long> fields = new HashSet<>();
         for (var entry : relaxData.entrySet()) {
             RelaxationData data = entry.getValue();
             fields.add(Math.round(data.getField()));
         }
         for (var field : fields) {
-            Double r1 = null, r1Error = null, r2 = null, r2Error = null, noe = null, noeError = null;
-            Double dField = null;
+            EnumMap<relaxTypes, RelaxationData> values = new EnumMap<>(relaxTypes.class);
             for (var entry : relaxData.entrySet()) {
                 RelaxationData data = entry.getValue();
-                if (Math.round(data.getField()) == field) {
-                    switch (data.getExpType()) {
-                        case R1:
-                            r1 = data.getValue();
-                            dField  = data.getField();
-                            r1Error = data.getError();
-                            break;
-                        case R2:
-                            r2 = data.getValue();
-                            r2Error = data.getError();
-                            break;
-                        case NOE:
-                            noe = data.getValue();
-                            noeError = data.getError();
-                            break;
-                    }
+                if (Math.round(data.getField()) == field && !data.getExpType().getName().startsWith("S") && data.getValue() != null) {
+                    values.put(data.getExpType(), data);
                 }
             }
-            R1R2NOE r1R2NOE = new R1R2NOE(r1, r1Error, r2, r2Error, noe, noeError, dField);
-            dataMap.put(field, r1R2NOE);
+            if (!values.isEmpty()) {
+                dataMap.put(field, values);
+            }
         }
         return dataMap;
     }
 
+    private static String toFileString(Map<relaxTypes, RelaxationData> values, List<relaxTypes> types) {
+        StringBuilder sBuilder = new StringBuilder();
+        for (var  type: types) {
+            var data = values.get(type);
+            if ((sBuilder.length() == 0) && (data != null)) {
+                sBuilder.append(String.format("%.2f", data.getField()));
+            }
+            Double value = data != null ? data.getValue() : null;
+            Double error = data != null ? data.getError() : null;
+            RelaxationValues.appendValueError(sBuilder, value, error,"%.3f");
+        }
+        return sBuilder.toString();
+    }
+
     public static void writeToFile(File file) throws IOException {
         MoleculeBase moleculeBase = MoleculeFactory.getActive();
```

Comment:
add comment here about why its unused so method doesn't get removed accidentally in future